packages/client/src/query/aggregations/AggregationResultsWithoutGroups.ts

```diff
@@ -16,25 +16,37 @@
 
 import type { ObjectOrInterfaceDefinition } from "@osdk/api";
 import type { OsdkObjectPropertyType } from "../../Definitions.js";
-import type { StringArrayToUnion } from "../../util/StringArrayToUnion.js";
-import type { AggregationClause } from "./AggregationsClause.js";
-
-type SubselectKeys<AC extends AggregationClause<any>, P extends keyof AC> =
-  AC[P] extends readonly string[] | string ? P : never;
+import type {
+  OrderedAggregationClause,
+  UnorderedAggregationClause,
+} from "./AggregationsClause.js";
 
 export type AggregationResultsWithoutGroups<
   Q extends ObjectOrInterfaceDefinition<any, any>,
-  AC extends AggregationClause<Q>,
+  AC extends UnorderedAggregationClause<Q> | OrderedAggregationClause<Q>,
 > = {
-  [P in keyof Q["properties"] as SubselectKeys<AC, P>]: AC[P] extends
-    readonly string[] | string ? {
-      [Z in StringArrayToUnion<AC[P]>]: Z extends "approximateDistinct" ? number
-        : OsdkObjectPropertyType<Q["properties"][P]>;
-    }
-    : never;
+  [
```

Comment:
This whole process of extracting a nested object structure from a flattened "property:metric" union is pretty messy. Is there a better way of doing this?

Reply:
Yeah so creating helper types starts to expose places where things can clean up. I ended up with this:

```ts
type ExtractPropName<T extends string> = T extends `${infer PREFIX}:${string}`
  ? PREFIX
  : T extends "$count" ? T
  : never;
type ExtractMetricForPropName<T, PropName extends string> = T extends
  `${PropName}:${infer METRIC_NAME}` ? METRIC_NAME : never;

export type AggregationResultsWithoutGroups<
  Q extends ObjectOrInterfaceDefinition<any, any>,
  AC extends UnorderedAggregationClause<Q> | OrderedAggregationClause<Q>,
> = {
  [TPropName in ExtractPropName<keyof AC & string>]: TPropName extends "$count"
    ? number
    : {
      [TMetricName in ExtractMetricForPropName<keyof AC & string, TPropName>]:
        TMetricName extends "approximateDistinct" ? number
          : (OsdkObjectPropertyType<Q["properties"][TPropName]>);
    };
};
```
I think this works but didn't check extensively.

A big difference here is I dropped the `as` in the mapped type and instead just directly returned the values, allowing me to reuse the propname and metricname.

Other places in the codebase where we have to do the `as` syntax, is because it allows us to retain things like jsdoc from the mapped types. However here we are mapping object literals that have no interesting information. `{ "foo:bar": true }` has no jsdoc to map, so we can simplify the syntax.

Now, if we wanted to some how keep the jsdoc descriptions of properties, we would have to map over the keys of Q["properties"] which would be doable. This is psuedocode I did not test:

```ts
export type AggregationResultsWithoutGroups<
  Q extends ObjectOrInterfaceDefinition<any, any>,
  AC extends UnorderedAggregationClause<Q> | OrderedAggregationClause<Q>,
> = {
  [ TPropName in keyof Q["properties"] ]: {
    [ K in ExtractMetricForPropName<keyof AC & string, TPropName>]: 
        TMetricName extends "approximateDistinct" ? number
          : (OsdkObjectPropertyType<Q["properties"][TPropName]>);
  }
  & {
    [ K in keyof AC as K extends "$count" ? K : never]: number;
};
```

Reply:
Tested this and it seems to work. I'll refactor this - thanks!

Reply:
I don't think we need the $count part in this because the $count gets &'d in with the AggregationCountResult type

Reply:
It seemed simpler to just remove the AggregationCountResult instead, so I just did that